src/sp/it/pl/util/functional/Util.java

```diff
@@ -420,47 +420,47 @@ static void repeat(int n, IntConsumer action) {
 		};
 	}
 
-	/** Faster alternative to {@link #noNull(java.lang.Object...) }. */
-	static <O> O noNull(O o1, O o2) {
+	/** Faster alternative to {@link #findNonNull(java.lang.Object...) }. */
+	static <O> O findNonNull(O o1, O o2) {
 		return o1!=null ? o1 : o2;
 	}
 
-	/** Faster alternative to {@link #noNull(java.lang.Object...) }. */
-	static <O> O noNull(O o1, O o2, O o3) {
+	/** Faster alternative to {@link #findNonNull(java.lang.Object...) }. */
+	static <O> O findNonNull(O o1, O o2, O o3) {
 		return o1!=null ? o1 : o2!=null ? o2 : o3;
 	}
 
-	/** Faster alternative to {@link #noNull(java.lang.Object...) }. */
-	static <O> O noNull(O o1, O o2, O o3, O o4) {
+	/** Faster alternative to {@link #findNonNull(java.lang.Object...) }. */
+	static <O> O findNonNull(O o1, O o2, O o3, O o4) {
 		return o1!=null ? o1 : o2!=null ? o2 : o3!=null ? o3 : o4;
 	}
 
 	/** Returns the first non null object or null if all null. */
 	@SafeVarargs
-	static <O> O noNull(O... objects) {
+	static <O> O findNonNull(O... objects) {
 		for (O o : objects)
 			if (o!=null) return o;
 		return null;
 	}
 
-	/** Faster alternative to {@link #noNull(java.util.function.Supplier...) }. */
-	static <I> I noNull(Supplier<I> supplier1, Supplier<I> supplier2) {
+	/** Faster alternative to {@link #findNonNull(java.util.function.Supplier...) }. */
+	static <I> I findNonNull(Supplier<I> supplier1, Supplier<I> supplier2) {
```

Comment:
I propose new name `firstNotNull`. Much clearer and does not imply 'searching'.

PS: it was a good idea to rename this method. The original name was ambiguous and overloaded with completely unrelated method.

Reply:
Yes, was also thinking about that. Will do.